lib/rules/prefer-user-event.ts

```diff
@@ -0,0 +1,138 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl, hasTestingLibraryImportModule } from '../utils';
+import { isImportSpecifier, isIdentifier, isMemberExpression } from '../node-utils'
+
+export const RULE_NAME = 'prefer-user-event'
+
+export type MessageIds = 'preferUserEvent'
+export type Options = [{ allowedMethods: string[] }];
+
+export const UserEventMethods = ['click', 'dblClick', 'type', 'upload', 'clear', 'selectOptions', 'deselectOptions', 'tab', 'hover', 'unhover', 'paste'] as const
+type UserEventMethodsType = typeof UserEventMethods[number]
+
+// maps fireEvent methods to userEvent. Those not found here, do not have an equivalet (yet)
+export const MappingToUserEvent: Record<string, UserEventMethodsType[]> = {
+  click: ['click', 'type', 'selectOptions', 'deselectOptions'],
+  change: ['upload', 'type', 'clear', 'selectOptions', 'deselectOptions'],
+  dblClick: ['dblClick'],
+  input: ['type', 'upload', 'selectOptions', 'deselectOptions', 'paste'],
+  keyDown: ['type', 'tab'],
+  keyPress: ['type'],
+  keyUp: ['type', 'tab'],
+  mouseDown: ['click', 'dblClick', 'selectOptions', 'deselectOptions'],
+  mouseEnter: ['hover', 'selectOptions', 'deselectOptions'],
+  mouseLeave: ['unhover'],
+  mouseMove: ['hover', 'unhover', 'selectOptions', 'deselectOptions'],
+  mouseOut: ['unhover'],
+  mouseOver: ['hover', 'selectOptions', 'deselectOptions'],
+  mouseUp: ['click', 'dblClick', 'selectOptions', 'deselectOptions'],
+  paste: ['paste'],
+  pointerDown: ['click', 'dblClick', 'selectOptions', 'deselectOptions'],
+  pointerEnter: ['hover', 'selectOptions', 'deselectOptions'],
+  pointerLeave: ['unhover'],
+  pointerMove: ['hover', 'unhover', 'selectOptions', 'deselectOptions'],
+  pointerOut: ['unhover'],
+  pointerOver: ['hover', 'selectOptions', 'deselectOptions'],
+  pointerUp: ['click', 'dblClick', 'selectOptions', 'deselectOptions'],
+}
+
+function buildErrorMessage(fireEventMethod: string) {
+  const allMethods = MappingToUserEvent[fireEventMethod].map((method: string) => `userEvent.${method}()`)
+
+  let userEventMethods = ''
+  if (allMethods.length > 2) {
+    userEventMethods += allMethods.slice(0, allMethods.length - 2).join(', ')
+  }
+
+  userEventMethods += `${allMethods.length > 1 ? ' or ' : ''}${allMethods[allMethods.length - 1]}`
+  return userEventMethods
```

Comment:
I think this is easier to read with variables extracted.
```suggestion
  const { length } = allMethods

  const init = length > 2 ? allMethods.slice(0, length - 2).join(', ') : ''
  const last = `${length > 1 ? ' or ' : ''}${allMethods[length - 1]}`
  return `${init}${last}`
```

Feel free to refactor more if you want, I may be missing something.